### PR TITLE
PS-7215 Release Notes for 5.7.31-34 (5.7)

### PR DESCRIPTION
**Pull request by patrickbirch**
modified:   source/conf.py
	new file:   source/release-notes/Percona-Server-5.7.31-34.rst
	modified:   source/release-notes/release-notes_index.rst

### CODE REVIEW

**Review by kathy-williamson — APPROVED** [review]

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+`Percona Server for MySQL <https://www.percona.com/software/mysql-database/percona-server>`_ 5.7.31-34
+includes all the features and bug fixes available in
+`MySQL 5.7.31 Community Edition <https://dev.mysql.com/doc/relnotes/mysql/5.7/en/news-5-7-31.html>`_
+in addition to enterprise-grade features developed by Percona.
+
+New Features
+================================================================================
+
+* :jirabug:`PS-7128`: Document RocksDB variables: rocksdb_max_background_compactions, rocksdb_max_background_flushes, and rocksdb_max_bottom_pri_background_compactions
+
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst:18]: use :variable: tags to link to variable descriptions

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+Improvements
+================================================================================
+
+* :jirabug:`PS-7132`: Make default value of rocksdb_wal_recovery_mode compatible with InnoDB
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst]: use :variable: tags to link to variable description

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+* :jirabug:`PS-7199`: Document Coredumper functionality
+* :jirabug:`PS-7114`: Enhance crash artifacts (core dumps and stack traces) to provide additional information to the operator
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst:26]: should this be "Implemented Coredumper funcionality", with link to feature documentation/description.

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+* :jirabug:`PS-7025`: Fix reading ahead of insert buffer pages by dispatching of buffered AIO transfers (Upstream :mysqlbug:`100086`)
+* :jirabug:`PS-7010`: Modify to Lock buffer blocks before sanity check in btr_cur_latch_leaves
+* :jirabug:`PS-6995`: Introduce a new optimizer switch to allow the user to reduce the cost of a range scan to determine best execution plan for Primary Key lookup
+* :jirabug:`PS-5978`: Remove unneeded check of variable to allow mysqld_safe support --numa-interleave (Thanks to user springlin for reporting this issue)
+* :jirabug:`PS-7220`: Fix activity counter update in purge coordinator and workers
+* :jirabug:`PS-7234`: Modify PS minimal tarballs to remove COPYING.AGPLv3
+* :jirabug:`PS-7204`: Add checks to linkingscript to correct failures in patchelf
+* :jirabug:`PS-7075`: Provide binary tarball with shared libs and glibc suffix
+* :jirabug:`PS-7062`: Modify ALTER INSTANCE ROTATE INNODB MASTER KEY to skip writing of redo for compressed encrypted temporary table.
+* :jirabug:`PS-5263`: Update handle_binlog_flush_or_sync_error() to set my_ok(thd) after thd->clear_error() to correct assert in THD::send_statement_status (Upstream :mysqlbug:`93770`)
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst:46]: this should be more user friendly description.

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+* :jirabug:`PS-7212`: Modify processing to binary compare in order to do native JSON comparison (Upstream :mysqlbug:`100307`)
+* :jirabug:`PS-7076`: Modify to not update Cardinality after setting tokudb_cardinality_scale_percent
+* :jirabug:`PS-7025`: Fix reading ahead of insert buffer pages by dispatching of buffered AIO transfers (Upstream :mysqlbug:`100086`)
+* :jirabug:`PS-7010`: Modify to Lock buffer blocks before sanity check in btr_cur_latch_leaves
+* :jirabug:`PS-6995`: Introduce a new optimizer switch to allow the user to reduce the cost of a range scan to determine best execution plan for Primary Key lookup
+* :jirabug:`PS-5978`: Remove unneeded check of variable to allow mysqld_safe support --numa-interleave (Thanks to user springlin for reporting this issue)
+* :jirabug:`PS-7220`: Fix activity counter update in purge coordinator and workers
+* :jirabug:`PS-7234`: Modify PS minimal tarballs to remove COPYING.AGPLv3
+* :jirabug:`PS-7204`: Add checks to linkingscript to correct failures in patchelf
+* :jirabug:`PS-7075`: Provide binary tarball with shared libs and glibc suffix
```
Comment (hrvojem) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst]: link to changed/updated docs with more details about the changes.

**Review by hrvojem — CHANGES_REQUESTED** [review]

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+* :jirabug:`PS-7132`: Make default value of rocksdb_wal_recovery_mode compatible with InnoDB
+* :jirabug:`PS-7199`: Document Coredumper functionality
+* :jirabug:`PS-7114`: Enhance crash artifacts (core dumps and stack traces) to provide additional information to the operator
+
+
+
+Bugs Fixed
+================================================================================
+
+* :jirabug:`PS-7203`: Modify open_tables to skip the test triggered if the thread originating it is a SQL thread or any replication thread (in the case of Multi thread replication)
```
Comment (george-lorch) [inline doc/source/release-notes/Percona-Server-5.7.31-34.rst]: This makes no sense and does not even mention the audit log plugin/functionality.  Please have @altmannmarcelo or @kathy-williamson re-write this to make sense.

**Review by george-lorch — APPROVED** [review]
Approved pending Hrvojes requests and my comment on PS-7203

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst (hunk 3 above)
Comment (patrickbirch) [reply doc/source/release-notes/Percona-Server-5.7.31-34.rst:26]: Changed to "Added".

File: doc/source/release-notes/Percona-Server-5.7.31-34.rst (hunk 6 above)
Comment (altmannmarcelo) [reply doc/source/release-notes/Percona-Server-5.7.31-34.rst]: @georgelorchpercona don't know where the above description came from. This is a leak on audit plugin with slave thread and those words should be mentioned. I changed it back to original title.

Comment (patrickbirch) [reply doc/source/release-notes/Percona-Server-5.7.31-34.rst]: Updated

**Review by hrvojem — APPROVED** [review]
LGTM

**Pull request merged**